src/Core/src/Handlers/WebView/WebViewHandler.Android.cs

```diff
@@ -1,39 +1,67 @@
 ﻿using Android.Webkit;
-using Android.Widget;
 using static Android.Views.ViewGroup;
 using AWebView = Android.Webkit.WebView;
 
 namespace Microsoft.Maui.Handlers
 {
 	public partial class WebViewHandler : ViewHandler<IWebView, AWebView>
 	{
+		WebViewClient? _webViewClient;
+		WebChromeClient? _webChromeClient;
+		bool _firstRun = true;
+
 		protected override AWebView CreateNativeView()
 		{
-			var aWebView = new MauiWebView(Context!)
+			return new MauiWebView(Context!)
 			{
-#pragma warning disable 618 // This can probably be replaced with LinearLayout(LayoutParams.MatchParent, LayoutParams.MatchParent); just need to test that theory
-				LayoutParameters = new AbsoluteLayout.LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent, 0, 0)
-#pragma warning restore 618
+				LayoutParameters = new LayoutParams(LayoutParams.MatchParent, LayoutParams.WrapContent)
 			};
+		}
 
-			if (aWebView.Settings != null)
-			{
-				aWebView.Settings.JavaScriptEnabled = true;
-				aWebView.Settings.DomStorageEnabled = true;
-			}
-			return aWebView;
+		public override void SetVirtualView(IView view)
+		{
+			base.SetVirtualView(view);
```

Comment:
Would it make sense to set `_firstRun = true` on line 22? And then just remove lines `25-26` ?

Reply:
done